pkg/controller/license/trial/trial_controller.go

```diff
@@ -95,7 +99,19 @@ func (r *ReconcileTrials) Reconcile(request reconcile.Request) (reconcile.Result
 	if err := r.reconcileTrialStatus(trialStatus); err != nil {
 		return reconcile.Result{}, pkgerrors.Wrap(err, "failed to reconcile trial status")
 	}
-	return reconcile.Result{}, nil
+	// 4. update trial secret if invalid to give user feedback
+	if r.isTrialRunning() && license.IsMissingFields() == nil {
+		verifier := licensing.Verifier{
+			PublicKey: r.trialPubKey,
+		}
+		status := verifier.Valid(license, time.Now())
+		if status != licensing.LicenseStatusValid {
+			secret.Annotations[licensing.LicenseInvalidAnnotation] = string(status)
+		}
+	} else {
+		secret.Annotations[licensing.LicenseInvalidAnnotation] = "trial can be started only once"
```

Comment:
I'm not sure I understand  the message "trial can be started only once" if `license.IsMissingFields()` is not `nil`

Reply:
This is supposed to give feedback in case a user had a trial at some point, then removed the trial license thereby reverting back to basic and _then_ creates a new empty trial secret (thus the missing fields) to start a new trial, which is not supported.

Reply:
I am adding a comment to make this clearer and maybe introduce a speaking variable

Reply:
Ok, thanks for the clarification.